quota/projectquota.go

```diff
@@ -291,15 +345,26 @@ func (q *Control) findNextProjectID(home string) error {
 			continue
 		}
 		path := filepath.Join(home, file.Name())
-		projid, err := getProjectID(path)
+		projid, err := checkProjID(path)
 		if err != nil {
 			return err
 		}
-		if projid > 0 {
-			q.quotas[path] = projid
+		if projid > 0 && projid != baseID {
+			continue
+		}
+		subfiles, err := ioutil.ReadDir(path)
```

Comment:
I'm not very familiar these paths.
How big can this be?

I have a feeling we should read the dir with a fixed buffer size.

Reply:
It's not too crazy, given that both overlay2 and local have another layer of subdirs below their top level dir, it's never going to actually reach a users FS to enumerate.
Also, this functions only runs once at daemon startup to build the initial state, and never at runtime.

Reply:
ping @cpuguy83 ptal ^